gdx/src/com/badlogic/gdx/scenes/scene2d/ui/Skin.java

```diff
@@ -142,6 +144,17 @@ public void remove (String name, Class type) {
 		if (type == Sprite.class) return (T)getSprite(name);
 
 		ObjectMap<String, Object> typeResources = resources.get(type);
+		if (typeResources == null) {
+			// Search within derived classes
+			Iterator<Class> typeIterator = resources.keys();
+			while (typeIterator.hasNext()) {
+				Class resourceType = typeIterator.next();
+				if (type.isAssignableFrom(resourceType)) {
+					typeResources = resources.get(resourceType);
+					break;
+				}
+			}
+		}

```

Comment:
I really like this. I have done something similar and created my own Skin and TextButton in my project just to be able to do this. I have many custom styles and it was a pain not being able to handle everything from json.